src/main/java/org/folio/rest/support/ShelvingOrderUpdate.java

```diff
@@ -0,0 +1,366 @@
+   *
+   * @param tx
+   * @param updatedRowsCount
+   * @param itemsUpdatePromise
+   * @return
+   */
+  private Future<Void> completeTransaction(Transaction tx, Integer updatedRowsCount,
+      Promise<Integer> itemsUpdatePromise) {
+    log.info("Invoking of completeTransaction, updatedRows size: {}", updatedRowsCount);
+
+    Future<Void> commitFuture = tx.commit()
+        .onSuccess(h -> itemsUpdatePromise.complete(updatedRowsCount))
+        .onFailure(h -> itemsUpdatePromise.fail(h.getCause().getMessage()));
+
+    log.info("Finishing of \"completeTransaction\", itemsUpdatePromise: {}, commitFuture: {}", itemsUpdatePromise, commitFuture);
+    return Future.succeededFuture();
+  }
+
+  /**
+   * Read items and fill them into data structures for further processing
+   *
+   * @param rowSet
+   * @return
+   */
+  public Future<List<Item>> aggregateRow2List(RowSet<Row> rowSet) {
+    log.info("Invoking of aggregateRow2List, rows size: {}", rowSet.size());
+    Promise<List<Item>> promise = Promise.promise();
+
+    if (rowSet.size() > 0) {
+      List<Item> targetList = new ArrayList<>();
+      rowSet.forEach(row -> {
+        JsonObject itemJsonObject = row.getJsonObject("itemJson");
+        log.info("Read JSON from database as JsonObject : {}", itemJsonObject);
+        log.info("JonObject toString: {}", itemJsonObject.toString());
+        Item item = itemJsonObject.mapTo(Item.class);
+        log.info("Deserialized item which was converted to JSON string : {}", item);
+        targetList.add(item);
+      });
+      promise.complete(targetList);
+    } else {
+      promise.complete(Collections.emptyList());
+    }
+
+    log.info("Finishing of aggregateRow2List, promise: {}", promise);
+    return promise.future();
+  }
+
+  /**
+   * Calculate items shelving order
+   *
+   * @return
+   */
+  public Future<Map<UUID, String>> calculateShelvingOrder(List<Item> itemList) {
+    log.info("Invoking of calculateShelvingOrder, itemList size: {}",
+        Optional.ofNullable(itemList).orElse(new ArrayList<>()).size());
+
+    final ObjectMapper objectMapper = new ObjectMapper();
+    Map<UUID, String> updatedItemsShelvingOrderMap = itemList.stream()
+        .map(EffectiveCallNumberComponentsUtil::getCalculateAndSetEffectiveShelvingOrder)
+        .collect(Collectors.toMap(
+            item -> UUID.fromString(item. getId()),
+            item -> {
+              log.info("map's value, an item: {}", item);
+              return item.getEffectiveShelvingOrder();
+            }));
+
+    log.info("Finishing of calculateShelvingOrder, updatedItemsShelvingOrderMap: {}", updatedItemsShelvingOrderMap);
+    return Future.succeededFuture(updatedItemsShelvingOrderMap);
+  }
+
+  /**
+   * Prepare items update parameters
+   *
+   * @param itemIdShelvingOrderMap
+   * @return
+   */
+  public Future<List<Tuple>> prepareItemsUpdate(Map<UUID, String> itemIdShelvingOrderMap) {
```

Comment:
No async operation is involved, this method should return `List<Tuple>`, no `Future` wrapping is needed.

Reply:
Done, thank you.